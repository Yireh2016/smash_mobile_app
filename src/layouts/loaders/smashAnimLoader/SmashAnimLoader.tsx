import * as React from 'react';
import { StyleSheet } from 'react-native';

import AnimatedLoader from '../../../components/animatedLoader/AnimatedLoader';
import { SMASHBLUE } from '../../../theme/theme';
import { H3, MainTitle } from '../../../components/texts/Texts';
import { Loader } from '../../../store/types/isLoading';

const styles = StyleSheet.create({
  lottie: {
    height: 200,
  },
  textAlignCenter: {
    textAlign: 'center',
  },
});

interface SmashAnimLoaderProps {
  loader: Loader | boolean;
}

const SmashAnimLoader = ({ loader }: SmashAnimLoaderProps) => {
  let isLoading;
  let _loader: Loader = { enable: false, title: '', message: '' };

  const isLoaderTypeBool = typeof loader === 'boolean';

  if (isLoaderTypeBool) {
    isLoading = loader;
  } else {
    _loader = typeof loader === 'object' ? { ...loader } : _loader;
    isLoading = _loader.enable;
  }

  const title =
    typeof _loader === 'object' && _loader && _loader.enable ? (
      <MainTitle style={styles.textAlignCenter}>{_loader?.title}</MainTitle>
    ) : null;

  const message =
    typeof _loader === 'object' && _loader && _loader.enable ? (
      <H3 style={styles.textAlignCenter}>{_loader?.message}</H3>
    ) : null;

  return (
    <AnimatedLoader
      visible={isLoading}
      overlayColor={SMASHBLUE}
      source={require('../../../assets/json/fullLoader.json')}
      animationStyle={styles.lottie}
      speed={1}
      title={title}
      message={message}
    />
  );
};
export default SmashAnimLoader;
