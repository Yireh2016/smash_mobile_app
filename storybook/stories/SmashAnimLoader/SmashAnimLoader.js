import React from 'react';

import { storiesOf } from '@storybook/react-native';

import { ThemeProvider } from 'styled-components';

import theme from '../../../src/theme/theme';
import { SafeAreaProvider } from 'react-native-safe-area-context';

//To check this on story book, please uncomment the add methods below
//this component is a fullsize loader that overlaps onto storybook tab navigator control

export default () =>
  storiesOf('SmashAnimLoader', module)
    .addDecorator((getStory) => (
      <SafeAreaProvider>{getStory()}</SafeAreaProvider>
    ))
    .addDecorator((getStory) => (
      <ThemeProvider theme={theme}>{getStory()}</ThemeProvider>
    ));
// .add('Custom loader', () => <SmashAnimLoader isLoading={true} />);
// .add('Custom loader', () => (
//   <SmashAnimLoader
//     isTest={true}
//     isLoading={true}
//     loader={{
//       enable: true,
//       message: 'Creating a personalized plan for Jainer...',
//       title: 'Working on your plan',
//     }}
//   />
// ));
