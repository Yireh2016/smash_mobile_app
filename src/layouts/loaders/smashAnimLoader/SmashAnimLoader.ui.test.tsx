import * as React from 'react';
import renderer from 'react-test-renderer';
import SmashAnimLoader from './SmashAnimLoader';

describe('SmashAnimLoader ui test', () => {
  it('SmashAnimLoader renders correctly boolean loader true', () => {
    const ComponentProps = {
      loader: true,
    };
    const Element = () => {
      return <SmashAnimLoader {...ComponentProps} />;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('SmashAnimLoader renders correctly boolean loader false', () => {
    const ComponentProps = {
      loader: false,
    };
    const Element = () => {
      return <SmashAnimLoader {...ComponentProps} />;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('SmashAnimLoader renders correctly loader object enable', () => {
    const ComponentProps = {
      loader: { enable: true, title: 'title', message: 'message' },
    };
    const Element = () => {
      return <SmashAnimLoader {...ComponentProps} />;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('SmashAnimLoader renders correctly loader object disable', () => {
    const ComponentProps = {
      loader: { enable: false, title: 'title', message: 'message' },
    };
    const Element = () => {
      return <SmashAnimLoader {...ComponentProps} />;
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
