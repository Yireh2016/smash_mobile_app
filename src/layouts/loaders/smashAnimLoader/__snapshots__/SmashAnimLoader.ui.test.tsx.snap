// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmashAnimLoader ui test SmashAnimLoader renders correctly boolean loader false 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={true}
  visible={false}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "bottom": 0,
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "#10202b",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "aspectRatio": 1,
            },
            undefined,
            Array [
              Object {
                "height": "100%",
                "width": "100%",
              },
              Object {
                "height": 200,
              },
            ],
          ]
        }
      >
        <View
          autoPlay={false}
          enableMergePathsAndroidForKitKatAndAbove={false}
          forwardedRef={[Function]}
          loop={true}
          onAnimationFinish={[Function]}
          onLayout={[Function]}
          progress={0}
          resizeMode="contain"
          sourceJson="{\\"v\\":\\"5.7.3\\",\\"fr\\":29.9700012207031,\\"ip\\":0,\\"op\\":60.0000024438501,\\"w\\":500,\\"h\\":500,\\"nm\\":\\"Comp 1\\",\\"ddd\\":0,\\"assets\\":[],\\"layers\\":[{\\"ddd\\":0,\\"ind\\":1,\\"ty\\":4,\\"nm\\":\\"Capa de formas 1\\",\\"sr\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":11},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":10},\\"p\\":{\\"a\\":0,\\"k\\":[250,250,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100],\\"ix\\":6}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"ind\\":0,\\"ty\\":\\"sh\\",\\"ix\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],\\"o\\":[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],\\"v\\":[[46.5,-183.5],[-3,-90],[-45,-120],[-56,-64],[-179,-110],[-102,-12],[-199,28],[-88,51],[-165,160],[-34,101],[6,189],[49,110],[206,168],[92,27],[193,-124],[71,-84]],\\"c\\":true},\\"ix\\":2},\\"nm\\":\\"Trazado 1\\",\\"mn\\":\\"ADBE Vector Shape - Group\\",\\"hd\\":false},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":0,\\"s\\":[0]},{\\"t\\":59.0000024031193,\\"s\\":[100]}],\\"ix\\":1},\\"e\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Recortar trazados 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\",\\"hd\\":false},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.152941176471,0.717647058824,0.509803921569,1],\\"ix\\":3},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":4},\\"w\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":0,\\"s\\":[5]},{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":30,\\"s\\":[12]},{\\"t\\":59.0000024031193,\\"s\\":[1]}],\\"ix\\":5},\\"lc\\":1,\\"lj\\":1,\\"ml\\":4,\\"bm\\":0,\\"nm\\":\\"Trazo 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\",\\"hd\\":false},{\\"ty\\":\\"fl\\",\\"c\\":{\\"a\\":0,\\"k\\":[0,0,0,1],\\"ix\\":4},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"r\\":1,\\"bm\\":0,\\"nm\\":\\"Relleno 1\\",\\"mn\\":\\"ADBE Vector Graphic - Fill\\",\\"hd\\":false},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transformar\\"}],\\"nm\\":\\"Forma 1\\",\\"np\\":4,\\"cix\\":2,\\"bm\\":0,\\"ix\\":1,\\"mn\\":\\"ADBE Vector Group\\",\\"hd\\":false}],\\"ip\\":0,\\"op\\":60.0000024438501,\\"st\\":0,\\"bm\\":0}],\\"markers\\":[]}"
          speed={1}
          style={
            Object {
              "aspectRatio": 1,
              "height": 200,
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`SmashAnimLoader ui test SmashAnimLoader renders correctly boolean loader true 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "bottom": 0,
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "#10202b",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "aspectRatio": 1,
            },
            undefined,
            Array [
              Object {
                "height": "100%",
                "width": "100%",
              },
              Object {
                "height": 200,
              },
            ],
          ]
        }
      >
        <View
          autoPlay={false}
          enableMergePathsAndroidForKitKatAndAbove={false}
          forwardedRef={[Function]}
          loop={true}
          onAnimationFinish={[Function]}
          onLayout={[Function]}
          progress={0}
          resizeMode="contain"
          sourceJson="{\\"v\\":\\"5.7.3\\",\\"fr\\":29.9700012207031,\\"ip\\":0,\\"op\\":60.0000024438501,\\"w\\":500,\\"h\\":500,\\"nm\\":\\"Comp 1\\",\\"ddd\\":0,\\"assets\\":[],\\"layers\\":[{\\"ddd\\":0,\\"ind\\":1,\\"ty\\":4,\\"nm\\":\\"Capa de formas 1\\",\\"sr\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":11},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":10},\\"p\\":{\\"a\\":0,\\"k\\":[250,250,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100],\\"ix\\":6}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"ind\\":0,\\"ty\\":\\"sh\\",\\"ix\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],\\"o\\":[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],\\"v\\":[[46.5,-183.5],[-3,-90],[-45,-120],[-56,-64],[-179,-110],[-102,-12],[-199,28],[-88,51],[-165,160],[-34,101],[6,189],[49,110],[206,168],[92,27],[193,-124],[71,-84]],\\"c\\":true},\\"ix\\":2},\\"nm\\":\\"Trazado 1\\",\\"mn\\":\\"ADBE Vector Shape - Group\\",\\"hd\\":false},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":0,\\"s\\":[0]},{\\"t\\":59.0000024031193,\\"s\\":[100]}],\\"ix\\":1},\\"e\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Recortar trazados 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\",\\"hd\\":false},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.152941176471,0.717647058824,0.509803921569,1],\\"ix\\":3},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":4},\\"w\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":0,\\"s\\":[5]},{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":30,\\"s\\":[12]},{\\"t\\":59.0000024031193,\\"s\\":[1]}],\\"ix\\":5},\\"lc\\":1,\\"lj\\":1,\\"ml\\":4,\\"bm\\":0,\\"nm\\":\\"Trazo 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\",\\"hd\\":false},{\\"ty\\":\\"fl\\",\\"c\\":{\\"a\\":0,\\"k\\":[0,0,0,1],\\"ix\\":4},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"r\\":1,\\"bm\\":0,\\"nm\\":\\"Relleno 1\\",\\"mn\\":\\"ADBE Vector Graphic - Fill\\",\\"hd\\":false},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transformar\\"}],\\"nm\\":\\"Forma 1\\",\\"np\\":4,\\"cix\\":2,\\"bm\\":0,\\"ix\\":1,\\"mn\\":\\"ADBE Vector Group\\",\\"hd\\":false}],\\"ip\\":0,\\"op\\":60.0000024438501,\\"st\\":0,\\"bm\\":0}],\\"markers\\":[]}"
          speed={1}
          style={
            Object {
              "aspectRatio": 1,
              "height": 200,
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`SmashAnimLoader ui test SmashAnimLoader renders correctly loader object disable 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={true}
  visible={false}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "bottom": 0,
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "#10202b",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "aspectRatio": 1,
            },
            undefined,
            Array [
              Object {
                "height": "100%",
                "width": "100%",
              },
              Object {
                "height": 200,
              },
            ],
          ]
        }
      >
        <View
          autoPlay={false}
          enableMergePathsAndroidForKitKatAndAbove={false}
          forwardedRef={[Function]}
          loop={true}
          onAnimationFinish={[Function]}
          onLayout={[Function]}
          progress={0}
          resizeMode="contain"
          sourceJson="{\\"v\\":\\"5.7.3\\",\\"fr\\":29.9700012207031,\\"ip\\":0,\\"op\\":60.0000024438501,\\"w\\":500,\\"h\\":500,\\"nm\\":\\"Comp 1\\",\\"ddd\\":0,\\"assets\\":[],\\"layers\\":[{\\"ddd\\":0,\\"ind\\":1,\\"ty\\":4,\\"nm\\":\\"Capa de formas 1\\",\\"sr\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":11},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":10},\\"p\\":{\\"a\\":0,\\"k\\":[250,250,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100],\\"ix\\":6}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"ind\\":0,\\"ty\\":\\"sh\\",\\"ix\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],\\"o\\":[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],\\"v\\":[[46.5,-183.5],[-3,-90],[-45,-120],[-56,-64],[-179,-110],[-102,-12],[-199,28],[-88,51],[-165,160],[-34,101],[6,189],[49,110],[206,168],[92,27],[193,-124],[71,-84]],\\"c\\":true},\\"ix\\":2},\\"nm\\":\\"Trazado 1\\",\\"mn\\":\\"ADBE Vector Shape - Group\\",\\"hd\\":false},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":0,\\"s\\":[0]},{\\"t\\":59.0000024031193,\\"s\\":[100]}],\\"ix\\":1},\\"e\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Recortar trazados 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\",\\"hd\\":false},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.152941176471,0.717647058824,0.509803921569,1],\\"ix\\":3},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":4},\\"w\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":0,\\"s\\":[5]},{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":30,\\"s\\":[12]},{\\"t\\":59.0000024031193,\\"s\\":[1]}],\\"ix\\":5},\\"lc\\":1,\\"lj\\":1,\\"ml\\":4,\\"bm\\":0,\\"nm\\":\\"Trazo 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\",\\"hd\\":false},{\\"ty\\":\\"fl\\",\\"c\\":{\\"a\\":0,\\"k\\":[0,0,0,1],\\"ix\\":4},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"r\\":1,\\"bm\\":0,\\"nm\\":\\"Relleno 1\\",\\"mn\\":\\"ADBE Vector Graphic - Fill\\",\\"hd\\":false},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transformar\\"}],\\"nm\\":\\"Forma 1\\",\\"np\\":4,\\"cix\\":2,\\"bm\\":0,\\"ix\\":1,\\"mn\\":\\"ADBE Vector Group\\",\\"hd\\":false}],\\"ip\\":0,\\"op\\":60.0000024438501,\\"st\\":0,\\"bm\\":0}],\\"markers\\":[]}"
          speed={1}
          style={
            Object {
              "aspectRatio": 1,
              "height": 200,
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`SmashAnimLoader ui test SmashAnimLoader renders correctly loader object enable 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "bottom": 0,
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "#10202b",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "width": 200,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 28,
            "textAlign": "center",
          }
        }
      >
        title
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "aspectRatio": 1,
            },
            undefined,
            Array [
              Object {
                "height": "100%",
                "width": "100%",
              },
              Object {
                "height": 200,
              },
            ],
          ]
        }
      >
        <View
          autoPlay={false}
          enableMergePathsAndroidForKitKatAndAbove={false}
          forwardedRef={[Function]}
          loop={true}
          onAnimationFinish={[Function]}
          onLayout={[Function]}
          progress={0}
          resizeMode="contain"
          sourceJson="{\\"v\\":\\"5.7.3\\",\\"fr\\":29.9700012207031,\\"ip\\":0,\\"op\\":60.0000024438501,\\"w\\":500,\\"h\\":500,\\"nm\\":\\"Comp 1\\",\\"ddd\\":0,\\"assets\\":[],\\"layers\\":[{\\"ddd\\":0,\\"ind\\":1,\\"ty\\":4,\\"nm\\":\\"Capa de formas 1\\",\\"sr\\":1,\\"ks\\":{\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":11},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":10},\\"p\\":{\\"a\\":0,\\"k\\":[250,250,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100,100],\\"ix\\":6}},\\"ao\\":0,\\"shapes\\":[{\\"ty\\":\\"gr\\",\\"it\\":[{\\"ind\\":0,\\"ty\\":\\"sh\\",\\"ix\\":1,\\"ks\\":{\\"a\\":0,\\"k\\":{\\"i\\":[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],\\"o\\":[[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0],[0,0]],\\"v\\":[[46.5,-183.5],[-3,-90],[-45,-120],[-56,-64],[-179,-110],[-102,-12],[-199,28],[-88,51],[-165,160],[-34,101],[6,189],[49,110],[206,168],[92,27],[193,-124],[71,-84]],\\"c\\":true},\\"ix\\":2},\\"nm\\":\\"Trazado 1\\",\\"mn\\":\\"ADBE Vector Shape - Group\\",\\"hd\\":false},{\\"ty\\":\\"tm\\",\\"s\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":0,\\"s\\":[0]},{\\"t\\":59.0000024031193,\\"s\\":[100]}],\\"ix\\":1},\\"e\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":2},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":3},\\"m\\":1,\\"ix\\":2,\\"nm\\":\\"Recortar trazados 1\\",\\"mn\\":\\"ADBE Vector Filter - Trim\\",\\"hd\\":false},{\\"ty\\":\\"st\\",\\"c\\":{\\"a\\":0,\\"k\\":[0.152941176471,0.717647058824,0.509803921569,1],\\"ix\\":3},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":4},\\"w\\":{\\"a\\":1,\\"k\\":[{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":0,\\"s\\":[5]},{\\"i\\":{\\"x\\":[0.667],\\"y\\":[1]},\\"o\\":{\\"x\\":[0.333],\\"y\\":[0]},\\"t\\":30,\\"s\\":[12]},{\\"t\\":59.0000024031193,\\"s\\":[1]}],\\"ix\\":5},\\"lc\\":1,\\"lj\\":1,\\"ml\\":4,\\"bm\\":0,\\"nm\\":\\"Trazo 1\\",\\"mn\\":\\"ADBE Vector Graphic - Stroke\\",\\"hd\\":false},{\\"ty\\":\\"fl\\",\\"c\\":{\\"a\\":0,\\"k\\":[0,0,0,1],\\"ix\\":4},\\"o\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"r\\":1,\\"bm\\":0,\\"nm\\":\\"Relleno 1\\",\\"mn\\":\\"ADBE Vector Graphic - Fill\\",\\"hd\\":false},{\\"ty\\":\\"tr\\",\\"p\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":2},\\"a\\":{\\"a\\":0,\\"k\\":[0,0],\\"ix\\":1},\\"s\\":{\\"a\\":0,\\"k\\":[100,100],\\"ix\\":3},\\"r\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":6},\\"o\\":{\\"a\\":0,\\"k\\":100,\\"ix\\":7},\\"sk\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":4},\\"sa\\":{\\"a\\":0,\\"k\\":0,\\"ix\\":5},\\"nm\\":\\"Transformar\\"}],\\"nm\\":\\"Forma 1\\",\\"np\\":4,\\"cix\\":2,\\"bm\\":0,\\"ix\\":1,\\"mn\\":\\"ADBE Vector Group\\",\\"hd\\":false}],\\"ip\\":0,\\"op\\":60.0000024438501,\\"st\\":0,\\"bm\\":0}],\\"markers\\":[]}"
          speed={1}
          style={
            Object {
              "aspectRatio": 1,
              "height": 200,
              "width": "100%",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "width": 216,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        message
      </Text>
    </View>
  </View>
</Modal>
`;
